### PR TITLE
[HttpFoundation][Session] Fix memcache session handler

### PR DESCRIPTION
**Pull request by klandaika**
| Q             | A
| ------------- | ---
| Branch?       | 2.1, 2.2, 2.3, 2.4, 2.5, 2.6, 2.7, 2.8, 3.0, 3.1, master
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT
| Doc PR        | 

Commit 0216e056051a6304cab01defc107fb583f3f5efa removed the opening of connection to memcached server on call to `open()`, because it's assumed that connection is already opened. However, `close()` still closes the connection. As a result no more read/write calls can be made if session got closed, as the connection does not get reestablished. 

Basically MemcacheSessionHandler should follow same logic as Memcache**d**SessionHandler, which is exactly what this MR acomplishes.

### CODE REVIEW

**Comment by iltar** [comment]
You should pick 2.7 as base branch for bug fixes, as the others are not supported any more. It could then possibly be merged upwards to 2.8, 3.1 and dev-master/3.2-dev

**Comment by klandaika** [comment]
Changed the base to 2.7 as requested.

**Comment by fabpot** [comment]
sounds reasonnable

**Comment by fabpot** [comment]
Thank you @klandaika.

**Comment by klandaika** [comment]
Do I have to do anything to get this merged up to master? or do you handle that internally?

**Comment by fabpot** [comment]
Nothing to do on your side, we merge old branches into newer ones on a regular basis. Thanks.
